Userland/Applications/Browser/History.cpp

```diff
@@ -13,25 +13,28 @@ void History::dump() const
     dbgln("Dump {} items(s)", m_items.size());
     int i = 0;
     for (auto& item : m_items) {
-        dbgln("[{}] {} {}", i, item, m_current == i ? '*' : ' ');
+        dbgln("[{}] {} '{}' {}", i, item.url, item.title, m_current == i ? '*' : ' ');
         ++i;
     }
 }
 
-void History::push(const URL& url)
+void History::push(const URL& url, String title)
 {
-    if (!m_items.is_empty() && m_items[m_current] == url)
+    if (!m_items.is_empty() && m_items[m_current].url == url)
         return;
     m_items.shrink(m_current + 1);
-    m_items.append(url);
+    m_items.append(History::URLTitlePair {
+        .url = url,
+        .title = title,
+    });
     m_current++;
 }
 
 URL History::current() const
 {
     if (m_current == -1)
         return {};
-    return m_items[m_current];
+    return m_items[m_current].url;
```

Comment:
Maybe we should return the complete pair instead? Seems a bit weird to store both and only return a URL for 'current'.

Reply:
I didn't want to break the signature of History::current(), and couldn't think of any use cases where anyone would need to use History::URLTitlePair from outside. Since it's only used in Tab.cpp it's an easy fix though